core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -139,15 +139,8 @@ trait WorkflowComponent {
           marshalInputResolution(inputResolution, workflowRecsByEntity(workflow.workflowEntity).id)
         }
 
-        val insertedRecQuery = for {
-          workflowRec <- findWorkflowsBySubmissionId(submissionId)
-          workflowEntityRec <- entityQuery if workflowEntityRec.id === workflowRec.workflowEntityId
-          insertedInputResolutionRec <- submissionValidationQuery if insertedInputResolutionRec.workflowId === workflowRec.id
-        } yield (workflowEntityRec, insertedInputResolutionRec)
```

Comment:
I don't really care that much, but why not just change what is yielded here? Did you need to go down the route of raw sql?

Reply:
I believe that doing it in the yield is too late - it still selects all columns in the underlying SQL. I did think of making a lighter drop-in replacement just for `entityQuery`, but then I saw that there were a handful of other columns we were selecting that weren't necessary, and decided to drop to raw SQL.

Reply:
Because at yield time it is still a Query and not yet an Action, Slick should reduce the data transfer to only what we need - as long as we do this before calling `result()` instead of after.

Reply:
My general preference is to keep it Slick-style, switching to Raw only if that's necessary.  So something like `yield (workflowEntityRec.entityType, workflowEntityRec.entityName, insertedInputResolutionRec)`